eth-bytecode-db/eth-bytecode-db/src/search/alliance_db.rs

```diff
@@ -107,3 +114,34 @@ fn extract_match_type(verified_contract_matches: &VerifiedContractMatches) -> Ma
         MatchType::Partial
     }
 }
+
+fn extract_libraries(
+    compiler_settings: &serde_json::Value,
+    verified_contract_matches: &VerifiedContractMatches,
+) -> Result<BTreeMap<String, String>, anyhow::Error> {
+    let mut libraries = solidity_libraries::try_parse_compiler_linked_libraries(compiler_settings)?;
+    match verified_contract_matches {
+        VerifiedContractMatches::OnlyCreation { creation_match } => {
+            libraries.extend(solidity_libraries::parse_manually_linked_libraries(
+                creation_match,
+            ));
+        }
+        VerifiedContractMatches::OnlyRuntime { runtime_match } => {
+            libraries.extend(solidity_libraries::parse_manually_linked_libraries(
+                runtime_match,
+            ));
+        }
+        VerifiedContractMatches::Complete {
+            creation_match,
+            runtime_match,
+        } => {
+            libraries.extend(solidity_libraries::parse_manually_linked_libraries(
+                creation_match,
+            ));
+            libraries.extend(solidity_libraries::parse_manually_linked_libraries(
+                runtime_match,
+            ));
+        }
+    }
```

Comment:
```suggestion
let all_matches = match verified_contract_matches {
        VerifiedContractMatches::OnlyCreation {
            creation_match: match_,
        }
        | VerifiedContractMatches::OnlyRuntime {
            runtime_match: match_,
        } => {
            vec![match_]
        }
        VerifiedContractMatches::Complete {
            creation_match,
            runtime_match,
        } => {
            vec![creation_match, runtime_match]
        }
    };
    libraries.extend(
        all_matches
            .into_iter()
            .flat_map(|m| solidity_libraries::parse_manually_linked_libraries(m)),
    );
```

this will remove multiple `parse_manually_linked_libraries` call repeats, but maybe it's much less readable. what do you think?

Reply:
I prefer explicit `parse_manually_linked_libraries` application for each match. But I've applied the combining of OnlyCreation and OnlyRuntime cases under one branch in [7da9cc9](https://github.com/blockscout/blockscout-rs/pull/1337/commits/7da9cc9f7dc80dda454c5e9658e9e0df9a6c1487. Thanks you, didn't know about that